src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -545,19 +547,40 @@ private async Task<DurableConfig> ReadDurableTaskConfig()
         // This is a stopgap approach to get the Durable extension version. It duplicates some logic in ExtensionManager.cs.
         private async Task<string> GetDurableMajorVersionAsync(JObject hostJson, ScriptJobHostOptions hostOptions)
         {
+            string metadataFilePath;
             bool isUsingBundles = hostJson != null && hostJson.TryGetValue("extensionBundle", StringComparison.OrdinalIgnoreCase, out _);
-            if (isUsingBundles)
+            // This feature flag controls whether to opt out of the SyncTrigger metadata fix for OOProc DF apps from https://github.com/Azure/azure-functions-host/pull/9331
+            if (FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableLegacyDurableVersionCheck))
             {
-                // TODO: As of 2019-12-12, there are no extension bundles for version 2.x of Durable.
-                // This may change in the future.
-                return "1";
-            }
+                // using legacy behavior, which concludes that out of process DF apps (including .NET isolated) are using DF Extension V1.x
+                // as a result, the SyncTriggers payload for these apps will be missing some metadata like "taskHubName"
+                if (isUsingBundles)
+                {
+                    return "1";
+                }
 
-            string binPath = binPath = Path.Combine(hostOptions.RootScriptPath, "bin");
-            string metadataFilePath = Path.Combine(binPath, ScriptConstants.ExtensionsMetadataFileName);
-            if (!FileUtility.FileExists(metadataFilePath))
+                string binPath = binPath = Path.Combine(hostOptions.RootScriptPath, "bin");
+                metadataFilePath = Path.Combine(binPath, ScriptConstants.ExtensionsMetadataFileName);
+                if (!FileUtility.FileExists(metadataFilePath))
+                {
+                    return null;
+                }
+            }
+            else
             {
-                return null;
+                if (isUsingBundles)
+                {
+                    // From Functions runtime V4 onwards, only bundles >= V2.x is supported, which implies the app should be using DF V2 or greater.
+                    return "2";
+                }
+
+                // If the app is not using bundles, we look for extensions.json
+                string metadataDirectoryPath = ScriptStartupTypeLocator.FindExtensionsMetadataPath(hostOptions.RootScriptPath, out _);
```

Comment:
Questioning the use of the static method on `ScriptStartupTypeLocator` from here. The alternative would be to move it to `Utility` (or some other more appropriate type).

OK without the change too.

Reply:
I had similar thoughts but I was trying to minimize changes unless explicitly requested :). Moved to `Utility`